skxray/img_proc/imgQuant.py

```diff
@@ -0,0 +1,135 @@
+# Module for the BNL image processing project
+# Developed at the NSLS-II, Brookhaven National Laboratory
+# Developed by Gabriel Iltis, Oct. 2013
+"""
+This module contains all of the tools which quantify segmented data sets.
+The typical input required for all of the tools is a labeled volume where
+each region or material of interest has been segmented and assigned an
+individual label (integer value).
+"""
+
+import numpy as np
+import scipy.ndimage.measurements as measure
+
+
+results_dict = {}
+def create_label_field(binary_vol_list,
+                       material_names_list,
+                       md_dict):
+    #md_dict should be optional input. If present then add to it.
+    return result_lblfield, mislabeled_voxel_vol, md_dict
+
+def Vol_trial_1(src_labelField,
+                num_bins, 
+                pd_function):
+    """
+    This function is a trial volume quantification tool which
+    generates  a histogram of the labeled volume. Each individual
+    material has its  own bin and corresponding voxel count. The
+    subsequent Q_VOL function  appears to be more readily useful, but
+    I'm retaining this function for  future development as histogram
+    analysis and comparison of features or  materials can be useful
+    for quantitative comparison of results.
+
+
+    Parameters
+    ----------
+    src_labelField : ndarray
+         The segmented data set converted to a label field
+
+    num_bins : integer

```

Comment:
So, do you think that num_bins should be renamed to something like num_phases, or num_materials? I need to rethink a standard definition for label_field, which will directly impact how the volume quantification functions will need to operate. Your suggestion, a long time ago, that materials within a label_field be assigned integer values according to 2^^Material_Number is the best way to address voxel assignment errors. However, using that format, num_bins (as it is currently included) would/could force the merging of bins, as you mention above.

I could simply define num_bins inside the function as:
num_bins = amax(label_field) - amin(label_field) + 1

That way there's no ambiguity, or room for merging of bins since every possible integer value assigned to a material gets its own bin.